projects/packages/waf/tests/php/integration/test-waf-activation.php

```diff
@@ -22,15 +25,56 @@ protected function set_up() {
 		// Set a blog token and id so the site is connected.
 		Jetpack_Options::update_option( 'blog_token', 'asdasd.123123' );
 		Jetpack_Options::update_option( 'id', 1234 );
+
+		// Set the WPCOM JSON API base URL so the site will attempt to make requests.
+		Constants::set_constant( 'JETPACK__WPCOM_JSON_API_BASE', 'https://public-api.wordpress.com' );
+	}
+
+	/**
+	 * Return a sample wpcom rules response.
+	 *
+	 * @return array
+	 */
+	public function return_sample_response() {
+		$sample_response = (object) array(
+			'data' => "<?php\n", // empty rules file
+		);
+
+		return array(
+			'body'     => wp_json_encode( $sample_response ),
+			'response' => array(
+				'code'    => 200,
+				'message' => '',
+			),
+		);
 	}
 
 	/**
 	 * Test WAF activation.
 	 */
 	public function testActivation() {
-		Waf_Initializer::on_activation();
+		// Mock the WPCOM request for retrieving the automatic rules.
+		add_filter( 'pre_http_request', array( $this, 'return_sample_response' ) );
```

Comment:
You may want to remove that filter at the end of the test, to avoid any potential issues in future tests.

Reply:
Thanks for catching this, added a `remove_filter` at the end of the test function here: https://github.com/Automattic/jetpack/pull/29005/commits/a80ff8dc72fa56dbacdaf1daaef3055edc16f9f0